tls/s2n_handshake_io.c

```diff
@@ -305,37 +305,56 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
     /* A handshake type has been negotiated */
     conn->handshake.handshake_type = NEGOTIATED;
 
+    if (conn->config->use_tickets) {
+        if (conn->session_ticket_status == S2N_DECRYPT_TICKET) {
+            if (!s2n_decrypt_session_ticket(conn, &conn->client_ticket_to_decrypt)) {
+                return 0;
+            }
+
+            conn->session_ticket_status = S2N_NEW_TICKET;
+            conn->handshake.handshake_type |= WITH_SESSION_TICKET;
+
+            /* If a session ticket is presented by the client, then skip lookup in Session ID server cache */
+            goto skip_cache_lookup;
+        }
+
+        if (conn->session_ticket_status == S2N_NEW_TICKET) {
+            conn->handshake.handshake_type |= WITH_SESSION_TICKET;
+        }
+    }
+
     /* If a TLS session is resumed, the Server should respond in its ServerHello with the same SessionId the
      * Client sent in the ClientHello. */
     if (s2n_allowed_to_cache_connection(conn) && !s2n_resume_from_cache(conn)) {
         return 0;
     }
 
-    if (conn->mode == S2N_CLIENT && conn->client_session_resumed == 1) {
-        return 0;
-    }
+    skip_cache_lookup:
+        if (conn->mode == S2N_CLIENT && conn->client_session_resumed == 1) {
```

Comment:
and all these subesquent indentation changes should be reverted

Reply:
Done.